File: C++/036. Valid Sudoku.cpp
class Solution {
public:
    bool CheckSudoku(int x,int y,vector<vector<char>>& board) {
        for(int i=0;i<9;i++)
        {
            if(y!=i && board[x][y]==board[x][i] || x!=i && board[x][y]==board[i][y])
            return false;
        }
        
        for(int i=0;i<3;i++)
        {
            for(int j=0;j<3;j++)
            {
                if(board[x][y]==board[x/3*3+i][y/3*3+j] && x!=x/3*3+i && y!=y/3*3+j){return false;}
            }
        }
        return true;
    }

    bool isValidSudoku(vector<vector<char>>& board) {
        for(int i=0;i<9;i++)
        {
            for(int j=0;j<9;j++)
            {
                if (board[i][j]=='.'){continue;}
                if (!CheckSudoku(i,j,board)){return false;}
            }
        }
        return true;
    }
};

File: python/036. Valid Sudoku.py
class Solution(object):
    def check(self, board, x, y):
        visited = {}
        for i in range(x, x+3):
            for j in range(y, y+3):
                if board[i][j] != "." and board[i][j] in visited: return False
                visited[board[i][j]] = True
        return True
    
    def isValidSudoku(self, board):
        """
        :type board: List[List[str]]
        :rtype: bool
        """
        
        for i in range(9):
            visited = {}
            for j in range(9):
                if board[i][j] != "." and board[i][j] in visited: return False
                visited[board[i][j]] = True
        for j in range(9):
            visited = {}
            for i in range(9):
                if board[i][j] != "." and board[i][j] in visited: return False
                visited[board[i][j]] = True
        for i in range(3):
            for j in range(3):
                if not self.check(board, 3 * i, 3 * j):
                    return False
        return True